### PR TITLE
Added required validation for Max dosage.

### LINKED ISSUE #2383: Add validation for max dosage in PRN prescription
image

Add backend validation to check that max dosage in 24 is greater than or equal to base dosage

https://github.com/coronasafe/care/blob/68a3a8281cf72eb859b5951ffbadda0399c3010f/care/facility/api/serializers/prescription.py#L97-L151

### PR DESCRIPTION
**Pull request by akat5uki**
## Proposed Changes

- Added validation to check if max dosage is less than base dosage. If that is the case, then raise validation error.

### Associated Issue

- Closes #2383

## Merge Checklist

- [ ] Tests added/fixed
- [X] Update docs in `/docs`
- [X] Linting Complete
- [X] Any other necessary step

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

File: care/facility/api/serializers/prescription.py [hunk 1]
```diff
@@ -127,6 +127,10 @@ def validate(self, attrs):
                 raise serializers.ValidationError(
                     {"indicator": "Indicator should be set for PRN prescriptions."}
                 )
+            if attrs.get("max_dosage") is None or attrs.get("max_dosage") < attrs.get("base_dosage"):
```
Comment (sainak) [inline care/facility/api/serializers/prescription.py]:
```suggestion
            if attrs.get("max_dosage") and attrs.get("max_dosage") < attrs.get("base_dosage"):
```

base_dosage is already validated on line 120, and max_dosage is non mandatory

**Review by sainak — CHANGES_REQUESTED** [review]
add test cases

File: care/facility/api/serializers/prescription.py (hunk 1 above)
Comment (akat5uki) [reply care/facility/api/serializers/prescription.py]: I have done the changes as suggested above.

**Review by sainak — CHANGES_REQUESTED** [review]
@akat5uki lint checks are failing, run precommit 

and add test cases for your change in https://github.com/coronasafe/care/blob/a4e2f8be3935060811e51141c237437298825a72/care/facility/tests/test_medicine_api.py#L9

**Comment by sainak** [comment]
read https://github.com/coronasafe/care/blob/develop/CONTRIBUTING.md

**Review by rithviknishad — APPROVED** [review]
validations lgtm; needs test cases and lint issue fixes as mentioned

**Comment by vigneshhari** [comment]
Closing because of inactivity

**Pull request closed without merging**